src/Adapter/Image/Uploader/AbstractImageUploader.php

```diff
@@ -107,29 +106,40 @@ protected function uploadFromTemp($temporaryImageName, $destination)
             throw new ImageOptimizationException('An error occurred while uploading the image. Check your directory permissions.');
         }
 
+        //@todo: should we care about tempfile deletion? shouldn't we leave it for garbage collector?
```

Comment:
Let the garbage collector do its job as it allows server admin to deal with it in the way he wants 😉 I dont think it's our code responsibility to clean up PHP tmp files

Reply:
Not sure if you noticed, but this class was released already, so would it mean BC break?
Regarding the BC breaks, could you also note that This AbstractImageUploader used to implement interface and I removed that (implemented interfaces in each class that extended this one instead). ts necessary if I want to reuse some of methods of this abstract class :thinking:

Reply:
If we are using `move_uploaded_file` php will take care of the deletion.
In theory - but needs to be checked in PHP doc https://www.php.net/manual/en/ - all tmp related php functions do the job. Some delete the file immediately like `move_uploaded_file` other do it when thread dies.

"When thread dies" can be an issue because it means your code might do a lot more thing before the file is deleted but it's edge-cases.